### DISCUSSION TITLE
Is it possible to clear FileUpload state?

### DISCUSSION THREAD

**Discussion by underdpt** [filamentphp/filament#8598] [Help]
Hello,

This is my use case: I have a FileUpload and when a file is added, I process it by using `afterStateUpdate` and show the file on another custom component (a basic table with extra information).

I would like to clear the uploaded files from the FileUpload component, but I've been unable to do so. I've tried to change the component state to null or [] on `afterStateUpdated` or `afterStateHydrated` but it doesn't work.

This is my component code:

```php
FileUpload::make('ficherosPropios')
    ->label('')
    ->reactive()
    ->disk('uploads')
    ->preserveFilenames()
    ->afterStateUpdated(function (
        Model $record,
        TemporaryUploadedFile $state,
        FileUpload $component,
        Pages\EditModel $livewire
    ) {
        try {
            $livewire->validateOnly('data.ficherosPropios');
        } catch (ValidationException $th) {
            $state->delete();

            throw $th;
        }

        if ($component->isDisabled()) {
            throw new Exception('Se ha superado el número de ficheros permitidos');
        }

        $nombreFichero = Str::of($state->getFilename())->beforeLast('.')->toString();
        $fichero       = $record->ficherosPropios()->create([
            'user_id'        => auth()->id(),
            'nombre'         => $state->getClientOriginalName(),
            'nombre_fichero' => $nombreFichero,
            'tipo'           => $state->getMimeType(),
            'size'           => $state->getSize(),
        ]);

        $result = rename(
            $state->getRealPath(),
            config('filesystems.disks.uploads.root') . '/' . $fichero->nombre_fichero
        );

        throw_unless($result, 'No se ha podido guardar el fichero');

        if ($record->ficherosPropios()->count() >= $record->empresa->max_num_ficheros) {
            $component->disabled(true);
        }

        $livewire->dispatch('refresh');
    })
    ->afterStateHydrated(function (FileUpload $component) {
        $component->state([]);
    })
```

Is this possible?

**Comment by leandrocfe** [comment]
I would say `$component->state(null)`

**Reply by underdpt** [reply]
Thanks,

Already tried, didn't work.

I think it's because this component uses `wire:ignore` so it's not rerendered.

**Reply by leandrocfe** [reply]
Can you share a project with this example on Github? Is it possible?

**Comment by gavinhewitt** [comment]
@underdpt Have you figured this out? Looking for the same :(

**Reply by underdpt** [reply]
No, I'm sorry i finally gave up due to time constraints and ended up reloading the page after every upload.

**Comment by jeffrey-kroonen** [comment]
I have solved the issue by doing the following:

Use one of the `after` hooks. In this example I have used the `afterSave()` hook. Then set the value of the desired field equal to `null` or an empty string, depending on your needs.

```php
<?php
class EditUser extends EditRecord
{
    ...
    protected function afterSave(): void
    {
        $this->data['logo'] = null;
    }
}
```

If you are using an create resource, you should be good by only using `$this->form->fill(['logo' => null])`

```php
<?php
class CreateUser extends CreateRecord
{
    ...
    protected function afterCreate(): void
    {
        $this->form->fill(['logo' => null]);
    }
}
```
